src/Interpreters/Cache/WriteBufferToFileSegment.cpp

```diff
@@ -114,7 +131,10 @@ std::unique_ptr<ReadBuffer> WriteBufferToFileSegment::getReadBufferImpl()
       * because in case destructor called without `getReadBufferImpl` called, data won't be read.
       */
     finalize();
-    return std::make_unique<ReadBufferFromFile>(file_segment->getPath());
+    if (file_segment->getDownloadedSize() > 0)
+        return std::make_unique<ReadBufferFromFile>(file_segment->getPath());
+    else
+        return std::make_unique<EmptyReadBuffer>();
```

Comment:
@kssenii Looks like this introduced this issue - https://github.com/ClickHouse/ClickHouse/issues/65877